app/src/main/java/org/glucosio/android/adapter/HistoryAdapter.java

```diff
@@ -26,44 +26,48 @@
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
-
+import org.glucosio.android.DomainConstants;
 import org.glucosio.android.R;
 import org.glucosio.android.presenter.HistoryPresenter;
 import org.glucosio.android.tools.GlucoseRanges;
 import org.glucosio.android.tools.GlucosioConverter;
+import org.glucosio.android.tools.NumberFormatUtils;
 
 import java.text.NumberFormat;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.List;
 
 public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.ViewHolder> {
     private final int metricId;
     private Context mContext;
     private ArrayList<String> weightDataTime;
     private ArrayList<Long> weightIdArray;
-    private ArrayList<Integer> weightReadingArray;
+    private List<Double> weightReadingArray;
     private ArrayList<String> ketoneDataTimeArray;
-    private ArrayList<Double> ketoneReadingArray;
+    private List<Double> ketoneReadingArray;
     private ArrayList<Long> ketoneIdArray;
     private ArrayList<String> pressureDateTimeArray;
-    private ArrayList<Integer> pressureMinArray;
-    private ArrayList<Integer> pressureMaxArray;
+    private List<Double> pressureMinArray;
+    private List<Double> pressureMaxArray;
     private ArrayList<Long> pressureIdArray;
     private ArrayList<String> cholesterolDateTimeArray;
-    private ArrayList<Integer> cholesterolHDLArray;
-    private ArrayList<Integer> cholesterolLDLArray;
-    private ArrayList<Integer> cholesterolTotalArray;
+    private List<Double> cholesterolHDLArray;
+    private List<Double> cholesterolLDLArray;
+    private List<Double> cholesterolTotalArray;
     private ArrayList<Long> cholesterolIdArray;
     private ArrayList<String> hb1acDateTimeArray;
     private ArrayList<Double> hb1acReadingArray;
     private ArrayList<Long> hb1acIdArray;
     private HistoryPresenter presenter;
     private ArrayList<Long> glucoseIdArray;
     private ArrayList<String> glucoseNotes;
-    private ArrayList<Integer> glucoseReadingArray;
+    private List<Double> glucoseReadingArray;
     private ArrayList<String> glucoseDateTime;
     private ArrayList<String> glucoseReadingType;
 
+    private NumberFormat numberFormat = NumberFormatUtils.createDefaultNumberFormat();
```

Comment:
As codacy-bot says, numberFormat should be final

Reply:
Fixed